parsing/parser.mly

```diff
@@ -1531,22 +1538,38 @@ structure_item:
         { pstr_typext $1 }
     | str_exception_declaration
         { pstr_exception $1 }
-    | module_binding
-        { $1 }
     | rec_module_bindings
         { pstr_recmodule $1 }
     | module_type_declaration
-        { let (body, ext) = $1 in (Pstr_modtype body, ext) }
-    | open_declaration
-        { let (body, ext) = $1 in (Pstr_open body, ext) }
+        { pstr_modtype $1 }
     | class_declarations
-        { let (ext, l) = $1 in (Pstr_class l, ext) }
+        { pstr_class $1 }
     | class_type_declarations
-        { let (ext, l) = $1 in (Pstr_class_type l, ext) }
+        { pstr_class_type $1 }
     | include_statement(module_expr)
         { pstr_include $1 }
     )
     { $1 }
+  | local_structure_item
+    { $1 }
+;
+
+(* A local structure item (= can appear in let expressions) *)
+local_structure_item:
+  | mkstr(
+      item_extension post_item_attributes
+        { let docs = symbol_docs $sloc in
+          Pstr_extension ($1, add_docs_attrs docs $2) }
+    )
```

Comment:
I don't understand why there is a special case for Pstr_extension, I wish it was handled like the rest below (or a comment to explain the difference), but this was already present before the PR.

Reply:
Actually, I removed this last commit because the same asymmetry exists for signatures. I will open a follow-up PR doing this cleanup everywhere.